src/types/api/sequencer.ts

```diff
@@ -362,10 +391,10 @@ export namespace Sequencer {
     };
     get_state_update: {
       QUERY: {
-        blockHash: string;
+        blockIdentifier: BlockIdentifier;
```

Comment:
`get_state_update` should support `{ blockHash, blockNumber }` as query parameters, with `blockNumber` accepting both block numbers/heights and the pending/latest tags ([link](https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/services/api/feeder_gateway/feeder_gateway_client.py#L146)). 

The tests pass on `devnet` because it ignores faulty parameters and returns as if the `latest` block was requested.